api/app/models/retro.rb

```diff
@@ -64,6 +66,22 @@ def validate_login?(val)
     end
   end
 
+  def validate_join_token?(val)
+    val == join_token
+  end
+
+  def is_magic_link_enabled=(val)
+    if ActiveModel::Type::Boolean.new.cast(val)
+      recompute_join_token
+    else
+      clear_join_token
+    end
+  end
```

Comment:
If the settings for the retro are updated, but the magic link param isn't touched, does this still cause it to recompute the join token?  Is that desired?

Reply:
Good catch. Yes it does, but it is not desired. Will fix